traversal/selector/matcher_util.go

```diff
@@ -0,0 +1,20 @@
+package selector
+
+import "io"
+
+type readerat struct {
+	io.ReadSeeker
+}
+
+// ReadAt provides the io.ReadAt method over a ReadSeeker.
+// This implementation does not support concurrent calls to `ReadAt`,
```

Comment:
So my one worry here is: what are the odds the consumer of this code reads this comment, and not the comment on golang's ReaderAt interface -- is there a chance for misuse through the chain of readerAt -> SectionReader -> NewBytesFromReader? SectionReader certainly seems to use ReadAt to implement seek+read for example. 

A seperate question: should AsLargeBytes return a ReadSeeker that is also a ReaderAt? os.File satisfies both. new section reader satisfies both.

Non blocking, but we should look at figuring this out when we can.

Reply:
@mvdan argued that `ReadSeeker` is a more common and lower effort to ask implementations of node to implement. This wrapper is internal (since the struct type is not exported) and so is only for use by the Slice matcher to wrap a `SectionReader` around a node's `ReadSeeker`. As such, I believe there's minimal potential for misuse.